intelmq-mailgen

```diff
@@ -145,12 +146,113 @@ def escape_sql_identifier(ident):
 
 
 def load_events(cur, columns, event_ids):
-    sql_columns = ", ".join(escape_sql_identifier(col) for col in columns)
-    cur.execute("SELECT {} FROM events WHERE id = ANY (%s)".format(sql_columns),
-                (event_ids,))
+    if columns:
+        sql_columns = ", ".join(escape_sql_identifier(col) for col in columns)
+        cur.execute("SELECT {} FROM events WHERE id = ANY (%s)".format(
+                    sql_columns),(event_ids,))
+    else:
+        cur.execute("SELECT * FROM events WHERE id = ANY (%s)", (event_ids,))
```

Comment:
Why the code duplication? The only difference between the branches is which columns are being selected, so something like the following would be clearer:

    if columns is not None:
        sql_columns = ", ".join(escape_sql_identifier(col) for col in columns)
    else:
        sql_columns = "*"
    cur.execute("SELECT {} FROM events WHERE id = ANY (%s)".format(sql_columns),
                (event_ids,))

The "is not None" seems better to me because the caller actually uses None to indicate that all columns should be fetched. Otherwise an empty columns list will also be interpreted as "all columns" when in fact it's probably an error.